inc/namespace.php

```diff
@@ -243,6 +260,181 @@ function get_cloudfront_distribution_config() : array {
+	$name = get_current_blog_id() . '-remap-host-header';
+	$function = $client->createFunction([
+		'FunctionCode' => file_get_contents( __DIR__ . '/cloudfront-function.js' ),
+		'FunctionConfig' => [
+			'Comment' => 'Sets the X-Original-Host header.',
+			'Runtime' => 'cloudfront-js-2.0',
+		],
+		'Name' => $name,
+	]);
+
+	$arn = $function['FunctionSummary']['FunctionMetadata']['FunctionARN'];
+	$etag = $function['ETag'];
+
+	$client->publishFunction([
+		'IfMatch' => $etag,
+		'Name' => $name,
+	]);
+
+	add_site_to_cloudfront_function( $arn, get_current_blog_id() );
+
+	update_option( 'hm-cloudfront-function', $arn );
+	return $arn;
+}
+
+/**
+ * Get all the created origin request policies on the network.
+ *
+ * We have to batch CloudFront Origin Request policies as we're limited by both the number of total policies and the number of
+ * associations between a single policy and many distributions.
+ *
+ * @return array<string, list<int>> A map of policy Id's to array of sites that are using the policy in their CloudFront Distributions.
+ */
+function get_cloudfront_functions() : array {
+	return get_site_option( 'hm-acm-cloudfront-functions', [] );
+}
+
+/**
+ * Add a site to the origin request policies on the network.
+ */
+function add_site_to_cloudfront_function( string $function_arn, int $site_id ) : void {
+	$policies = get_cloudfront_functions();
+	$policies[ $function_arn ][] = $site_id;
+	update_site_option( 'hm-acm-cloudfront-functions', $policies );
+}
+
+
+function has_cloudfront_origin_request_policy() : bool {
+	return get_option( 'hm-cloudfront-origin-request-policy', false );
+}
+
+function unlink_cloudfront_origin_request_policy() {
+	delete_option( 'hm-cloudfront-origin-request-policy' );
+}
+
+function get_cloudfront_origin_request_policy_id(): ?string {
+	return get_option( 'hm-cloudfront-origin-request-policy', null );
+}
+/**
+ * Create or find an existing CloudFront Origin Request policy.
+ *
+ * @return string The Origin Request Policy Id.
+ *
+ */
+function create_cloudfront_origin_request_policy() : string {
+
+	$existing_policies = get_cloudfront_origin_request_policies();
+	foreach ( $existing_policies as $policy_id => $sites ) {
+		if ( count( $sites ) >= HM_ACM_ORIGIN_REQUEST_POLICIES_PER_DISTRIBUTION ) {
+			continue;
+		}
+
+		update_option( 'hm-cloudfront-origin-request-policy', $policy_id );
+		add_site_to_cloudfront_origin_request_policy( $policy_id, get_current_blog_id() );
+		return $policy_id;
+	}
+
+	$client = get_aws_cloudfront_client();
+	$name = get_current_blog_id() . '-hm-acm';
+	$policy = $client->createOriginRequestPolicy([
+		'OriginRequestPolicyConfig' => [
+			'Comment' => 'HM-ACM origin request policy',
+			'Name' => $name,
+			'HeadersConfig' => [
+				'HeaderBehavior' => 'allViewer',
+			],
+			'QueryStringsConfig' => [
+				'QueryStringBehavior' => 'all',
+			],
+			'CookiesConfig' => [
+				'CookieBehavior' => 'whitelist',
+				'Cookies' => [
+					'Quantity' => 3,
+					'Items' => [
```

Comment:
Could the cookies list be made filterable to allow for custom cookies to be added to the list when the request policies are created?

I know if that list does ever change then we'd need to either manually update the policy via AWS console or possibly introduce another function for updating policies using the plugin for that use case. Just thinking about that first time policy creation given there can potentially be multiple policies to be created and would be good to have other cookies already included.